core/arch/arm/kernel/generic_boot.c

```diff
@@ -552,6 +555,37 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_pseudo_device_node(void *fdt)
+{
+	char s[33];
+	int offs, n = 0, m, len;
+	const uint32_t *p;
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "pseudo-device-id", &len);
```

Comment:
- I'd rather drop the "pseudo-`. IMO it adds no value.
- Since it's a node that is read by secure world only, we may want to add the "secure-" prefix. Again I'm refering to this kernel commit: [d9f43ba](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=d9f43babb998b4061305ec682a423c709a503178) ("Documentation: dt: Add bindings for Secure-only devices")
- I'd like this binding to be documented somewhere. That's actually a general question for all subnodes of `/firmware/optee`: where should they be documented? I'd say: data consumed by non-secure world should be documented with the kernel bindings, and the secure ones should go somewhere in `optee_os/documentation`. Thoughts?
- There's a typo in your commit message ('psuedo').